stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/MessageDeletedViewHolder.kt

```diff
@@ -0,0 +1,36 @@
+package io.getstream.chat.android.ui.messages.adapter.viewholder
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import com.getstream.sdk.chat.adapter.MessageListItem
+import io.getstream.chat.android.ui.databinding.StreamItemMessageDeletedBinding
+import io.getstream.chat.android.ui.messages.adapter.BaseMessageItemViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.decorator.BackgroundDecorator
+import io.getstream.chat.android.ui.messages.adapter.viewholder.decorator.Decorator
+import io.getstream.chat.android.ui.messages.adapter.viewholder.decorator.GapDecorator
+
+public class MessageDeletedViewHolder(
+    parent: ViewGroup,
+    internal val binding: StreamItemMessageDeletedBinding = StreamItemMessageDeletedBinding.inflate(
+        LayoutInflater.from(
+            parent.context
+        ),
+        parent,
+        false
+    )
+) : BaseMessageItemViewHolder<MessageListItem.MessageItem>(binding.root) {
+
+    private val decorators = listOf<Decorator>(BackgroundDecorator(), GapDecorator())
+
+    override fun bind(data: MessageListItem.MessageItem) {
+        decorators.forEach { it.decorate(this, data) }
```

Comment:
Perhaps this behaviour could be pushed into a base class too, the support for decorators. But it can also wait until later.

Reply:
Will move that later in other PR